src/main/kotlin/com/katanox/tabour/integration/sqs/core/consumer/SqsEventPoller.kt

```diff
@@ -17,69 +32,96 @@ import java.util.concurrent.TimeUnit
 private val logger = KotlinLogging.logger {}
 
 class SqsEventPoller(
-    private val name: String,
+    private val queueUrl: String,
     private val eventHandler: SqsEventHandler,
-    private val eventFetcher: SqsEventFetcher,
-    private val pollerThreadPool: ScheduledThreadPoolExecutor,
-    private val handlerThreadPool: ThreadPoolExecutor,
-    private val pollingProperties: EventPollerProperties,
-    private val sqsConfiguration: SqsConfiguration,
-    private val exceptionHandler: ExceptionHandler
+    private val client: SqsClient,
+    private val pollerConfigs: EventPollerProperties,
+    private val tabourConfigs: TabourAutoConfigs,
 ) {
+
+    private lateinit var job: Job
+
+    @OptIn(ExperimentalTime::class, ExperimentalCoroutinesApi::class, DelicateCoroutinesApi::class)
     fun start() {
-        logger.info("starting SqsMessagePoller")
-        for (i in 0 until pollerThreadPool.corePoolSize) {
-            logger.info("starting SqsMessagePoller ({}) - thread {}", name, i)
-            pollerThreadPool.scheduleWithFixedDelay(
-                { pollMessages() },
-                pollingProperties.pollDelay.seconds,
-                pollingProperties.pollDelay.seconds,
-                TimeUnit.SECONDS
-            )
+        logger.debug("starting SqsMessagePoller ${LocalDateTime.now()}")
+        job = GlobalScope.launch {
```

Comment:
this can be changed to 

```kotlin
suspend fun start() = launch {
  //...
}
```
with the changes mentioned above and you can remove the `GlobalScope.launch`

Reply:
The reason that it was assigned to a job is to make it possible to cancle